### DISCUSSION TITLE
Test application service methods with or without authorization

### DISCUSSION THREAD

**Discussion by cephaspad** [abpframework/abp#10444] [Q&A]
May anyone guide me to Application Service Methods with two following cases?

1. Test a function when user is signed in.
2. Test a function when user has not yet signed in.

Appkication:
```
public class SomethingAppService : OurApplicationService
{
   [Authorize]
   public string GetDisplayName()
   {
      return base.CurrentUser.GetName();
   }
}
```

Thanks.

**Comment by maliming** [comment]
You can simulate the current user before testing the method.
https://github.com/abpframework/abp/blob/dev/framework/test/Volo.Abp.AspNetCore.Mvc.Tests/
https://github.com/abpframework/abp/blob/dev/framework/test/Volo.Abp.AspNetCore.Mvc.Tests/Volo/Abp/AspNetCore/Mvc/Authorization/AuthTestController_Tests.cs

**Comment by cephaspad** [comment]
Is there any other way? I develop app on BlazorWasm, then there is no AspNetCoreMvcTestBase.

Thanks.

**Reply by maliming** [reply]
You can check the [bunit](https://bunit.dev/docs/test-doubles/faking-auth.html)

**Reply by cephaspad** [reply]
thanks

**Comment by cephaspad** [comment]
Is there any way for me to use NSubtitute?
